src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandIncludeProperty.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData.Query.Expressions
+{
+    internal class SelectExpandIncludeProperty
+    {
+        /// <summary>
+        /// the corresponding property segment.
+        /// </summary>
+        private PropertySegment _propertySegment;
+
+        /// <summary>
+        /// the corresponding navigation source. maybe useless
+        /// </summary>
+        private IEdmNavigationSource _navigationSource;
+
+        /// <summary>
+        /// the path select item for this property.
+        /// for example: $select=abc or $select=NS.Type/abc
+        /// </summary>
+        private PathSelectItem _propertySelectItem;
+
+        /// <summary>
+        /// the sub $select and $expand for this property.
+        /// </summary>
+        private IList<SelectItem> _subSelectItems;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="SelectExpandIncludeProperty"/> class.
+        /// </summary>
+        /// <param name="propertySegment">The property segment that has this select expand item.</param>
+        public SelectExpandIncludeProperty(PropertySegment propertySegment)
+            : this(propertySegment, null)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="SelectExpandIncludeProperty"/> class.
+        /// </summary>
+        /// <param name="propertySegment">The property segment that has this select expand item.</param>
+        /// <param name="navigationSource">The targe navigation source of this property segment.</param>
+        public SelectExpandIncludeProperty(PropertySegment propertySegment, IEdmNavigationSource navigationSource)
+        {
+            if (propertySegment == null)
+            {
+                throw Error.ArgumentNull("propertySegment");
+            }
+
+            _propertySegment = propertySegment;
+            _navigationSource = navigationSource;
+        }
+
+        /// <summary>
+        /// Gets the merged path select item for this property, <see cref="PathSelectItem"/>.
+        /// </summary>
+        /// <returns>Null or the created <see cref="PathSelectItem"/>.</returns>
+        public PathSelectItem ToPathSelectItem()
+        {
+            if (_subSelectItems == null)
```

Comment:
>_subSelectItems [](start = 16, length = 15)

change to this._subSelectItems for clarity/consistency?

Reply:
I prefer to "not use this." for underscore members. So, i remove the other places using "this.".

---
In reply to: [355637081](https://github.com/OData/WebApi/pull/1941#discussion_r355637081) [](ancestors = 355637081)